### PR TITLE
Email service

### PR DESCRIPTION
**Pull request by anaturrillo**
## Summary

I added a welcome email sending to the user creation process:
Feature deployed at: https://at-wolox-trainning.herokuapp.com/users (POST)

## Trello Card

https://trello.com/c/uu8PtymQ/37-mail-de-bienvenida

### CODE REVIEW

File: test/sigunp.spec.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 const supertest = require('supertest');
-const { expect } = require('chai');
+const { expect: expectChai } = require('chai');
```
Comment (mpicollo-wolox) [inline test/sigunp.spec.js]: Why do you rename this to `expectChai`?

File: test/email.spec.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const supertest = require('supertest');
+const app = require('../app');
+const request = supertest(app);
+const { mockUser } = require('./mocks/mockUsers');
+const emails = require('../app/services/emails');
+jest.mock('../app/services/emails');
```
Comment (mpicollo-wolox) [inline test/email.spec.js]: This should me mock in signup test too

File: app/templates/emails.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+exports.welcomeEmail = user => ({
+  from: 'SuperDooper Api',
```
Comment (mpicollo-wolox) [inline app/templates/emails.js]: You don't need a email address here?

File: test/email.spec.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+describe('Email sending', () => {
+  test('Welcome email on signup', () => {
```
Comment (mpicollo-wolox) [inline test/email.spec.js]: Maybe move this test to signup suite

File: test/app.spec.js [hunk 5]
```diff
@@ -3,6 +3,10 @@ const fs = require('fs'),
   models = require('../app/models'),
   path = require('path');
 
+const emails = require('../app/services/emails');
+jest.mock('../app/services/emails');
+emails.sendWelcomeEmail.mockResolvedValue('SEND EMAIL');
```
Comment (mpicollo-wolox) [inline test/app.spec.js]: This is necessary here ?

Comment (anaturrillo) [reply test/app.spec.js]: Without it the mocking only worked when running the specific test. I couldn't find another way to make this work.

File: test/sigunp.spec.js (hunk 1 above)
Comment (anaturrillo) [reply test/sigunp.spec.js]: I initially started the email tests in the signup suit. But when I run into some issues I created a suite for email testing (I will reintegrated it here though) and forgot to remove it. So.. to check that the email function has been called I'm using jest's expect, so I renamed chai's expect to keep using both. I did not realize initially I could simply use jest's expect everywhere.

File: test/app.spec.js (hunk 5 above)
Comment (mpicollo-wolox) [reply test/app.spec.js]: This should be only necessary in specific test suite, is repeated in test file too

File: test/sigunp.spec.js (hunk 1 above)
Comment (mpicollo-wolox) [reply test/sigunp.spec.js]: I realized now that you are using chai expect in test suites, you should use jest integrated expect (change it only in this new test suit or in signup only)

File: test/app.spec.js [hunk 6]
```diff
@@ -3,6 +3,8 @@ const fs = require('fs'),
   models = require('../app/models'),
   path = require('path');
 
+jest.mock('../app/services/emails');
```
Comment (mpicollo-wolox) [inline test/app.spec.js]: Mm i don't like have all mocks in app.spec.js, can we try something like this in signup test ? 
`emailService.sendMail = jest.fn().mockResolvedValueOnce('Nice');`

File: app/templates/emails.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+const { email } = require('../../config').common;
+
+exports.welcomeEmail = user => ({
+  from: `"SuperDooper Api" <${email.auth.user}>`,
```
Comment (mpicollo-wolox) [inline app/templates/emails.js]: This should be something like `no-reply@api.com.co` (as a config variable)

**Pull request merged**